onnxruntime/test/ep_graph/test_ep_graph.cc

```diff
@@ -449,6 +449,24 @@ static void CheckGraphCApi(const GraphViewer& graph_viewer, const OrtGraph& api_
 
     CheckValueInfosCApi(graph_viewer, api_node_outputs, output_node_args);
 
+    // Check node attributes
+    const auto& node_attrs = node->GetAttributes();
+
+    if (node_attrs.size() > 0) {
+      OrtArrayOfConstObjects* api_node_attributes = nullptr;
+      DeferOrtRelease<OrtArrayOfConstObjects> release_node_attributes(&api_node_attributes,
+                                                                      ort_api.ReleaseArrayOfConstObjects);
+      ASSERT_ORTSTATUS_OK(ort_api.Node_GetAttributes(api_node, &api_node_attributes));
+      CheckArrayObjectType(api_node_attributes, ORT_TYPE_TAG_OrtOpAttr);
+
+      for (size_t attr_idx = 0; attr_idx < node_attrs.size(); attr_idx++) {
+        const OrtOpAttr* api_node_attr = nullptr;
+        ASSERT_ORTSTATUS_OK(ort_api.ArrayOfConstObjects_GetElementAt(api_node_attributes, attr_idx,
+                                                                     reinterpret_cast<const void**>(&api_node_attr)));
+        ASSERT_NE(api_node_attr, nullptr);
```

Comment:
I was about to suggest checking that the attribute retrieved from the API has the same type as the attribute from the internal graph (no need to check contents). However, I don't think we have a C API to get the `OrtOpAttrType` from an `OrtOpAttr`. I think that API would be necessary for an EP author, right? Could that be added in this PR?

Reply:
That's a good point and user might need to know the attribute type.
Just wondering if we really need to add a C API to get the type of OrtOpAttr? Since OrtOpAttr is interchangeable with ONNX::AttributeProto, and user can cast it to AttributeProto and call type() to get the ONNX::AttributeProto_AttributeType
Or we shouldn't assume users will add ONNX as a dependency?

Reply:
I think for normal users of the API we shouldn't assume or require them to cast to onnx::AttributeProto.

Reply:
OpAttr_GetType C API is added